src/Microsoft.VisualStudio.Threading/DispatcherExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿/********************************************************
+*                                                        *
+*   © Copyright (C) Microsoft. All rights reserved.      *
+*                                                        *
+*********************************************************/
+
+#if NET45
```

Comment:
I would expect the scenario to work in both directions, through one of the following approaches:

1. The .NET Standard library is actually a reference assembly, such that it's impossible to actually cause that image to be loaded for execution at runtime. Only one image can actually be executed, and that image works for both users who compiled against the net45 and users who compiled against .NET Standard.
2. The .NET 4.5 specific items are actually placed in a second assembly which lives alongside the main one, such as Microsoft.VisualStudio.Threading.Desktop. This is the approach taken by Roslyn workspaces desktop features, for example.